python/selfie-lib/selfie_lib/PerCharacterEscaper.py

```diff
@@ -45,41 +41,43 @@ def escape(self, input_string: str) -> str:
             return "".join(result)
 
     def unescape(self, input_string: str) -> str:
-        if input_string.endswith(
-            chr(self.__escape_code_point)
-        ) and not input_string.endswith(chr(self.__escape_code_point) * 2):
-            raise ValueError(
-                f"Escape character '{chr(self.__escape_code_point)}' can't be the last character in a string."
-            )
-
-        no_escapes = self.__first_offset_needing_escape(
-            input_string, self.__escape_code_point
-        )
-        if no_escapes == -1:
```

Comment:
I think the old `unescape` method is much better than this new one.

Most strings don't need to be escaped. In the "before" version, these strings are processed very fast because we don't need to allocate the result array.

In the "after" version, we do the entire slow character-by-character copy, and only after we've done all that work do we check to see if we can just return the original string after all. That defeats the purpose of returning the original string.

In the "after" version, there is no speedup - you compute the full thing, and return the

Reply:
I have switched back to the before version logic.